DESIGN.md

```diff
@@ -0,0 +1,52 @@
+Warnet is a software suite that allows its users to safely put ideas about Bitcoin to the test. It does this by using Kubernetes to simulate Bitcoin networks and providing user interaction. This document explains how Warnet does this by using design philosophies popular with Kubernetes such as Infrastructure as Code and a tendency towards Stateless Configuration and Externalized Configuration.
+
+# Project structure
+The Warnet code base has four main sections:
+
+1. resources - these items are available during runtime and relate to configuration (crucially, Kubernetes configuration)
+2. src/warnet - python source code lives here
+3. test - CI testing files live here
+4. docs - stores documentation available in the github repository
+
+## Overview of resources
+There are four main kinds of *resources*:
+
+1. Kubernetes configuration files - they are the backbone of Stateless Configuration; they are *yaml* files
+2. scenarios - these are python programs that users can load into Kubernetes to interact with the simulated Bitcoin network
+3. images - the logic for creating bitcoin nodes and also containers for running scenarios are found here; this includes Dockerfiles
+4. scripts and other configs - these are like "assets" or "one off" items which appear in Warnet.
+
+## Overview of src/warnet
+The python source code found in *src/warnet* serves to give users a way to create and interact with the simulated Bitcoin network as it exists in a Kubernetes cluster.
+
+There are eight categories of python program files in Warnet:
+
+1. Bitcoin images
+  * image.py and image_build.py - the logic that helps the user create bitcoin node images
+2. Bitcoin interaction
+  * bitcoin.py - make it easy to interact with bitcoin nodes in the simulated network
+3. Scenario interaction
+  * control.py - launch scenarios in order to interact with the simulated Bitcoin network
+4. Kubernetes
+  * k8s.py - gather Kubernetes configuration data; retrieve Kubernetes resources
+  * status.py - make it easy for the user to see the status of the simulated bitcoin network
+5. Resource configuration pipeline
+  * admin.py - copy configurations for *resources* such as namespaces and put them in the user's directory
+  * deploy.py - take configurations for *resources* and put them into the Kubernetes cluster
+  * network.py - copy *resources* to the users Warnet directory
+  * namespaces.py - copy *resources* to the users Warnet directory; interact with namespaces in the cluster
+6. User interaction
+  * main.py - provide the interface for the `warnet` command line program
+7. Host computer
+  * process.py - provides a way to run commands on the user's host computer
+8. Externalized configuration
+  * constants.py - this holds values which occur repeatedly in the code base
+
+## Overview of test
+The test_base.py file forms the basis of the *test* section. Each test uses *TestBase* which in turn uses the original Bitcoin Test Framework which can be found in the *src* directory.
```

Comment:
test_base does *not* inherit the test framework

Reply:
yeah this is the `Commander` class